Fix custom property assignment types

Th3Gavst3r opened the pull request:
Modifies the `Object Action Output` node's attribute setter function to perform a type cast.

## Notes
* `object.bl_rna.properties` does not contain Custom Properties, but copies of items in `bl_rna.properties` seem to exist directly on the `object` (such as `object.bl_rna.properties[location]` and `object.location`).
* Unlike `bl_rna.properties`, properties accessed directly from the `object` do not have an `array_length` attribute, so `len()` was used instead.

Resolves #1815

OmarEmaraDev commented:
Sorry for the delay, I have a lot on my plate but and will get to this as soon as possible.

OmarEmaraDev commented on animation_nodes/utils/attributes.py:
```diff
@@ -20,19 +20,24 @@ def getAttributeGetter(propName):
-        return '{}.{} = {}'.format(objectName, propName, valueName)
+        path = "{}.{}".format(objectName, propName)
+
+    return (
```
Use `yield` instead of return. Double quotes instead of singles quotes. Comments go above statements. Use f-strings.

```python
# Try to cast to existing property type.
yield f"try: {path} = type({path})({valuename})"
# Property does not exist, default to new float.
yield f"except: {path} = {valueName}"
```

OmarEmaraDev commented on animation_nodes/utils/attributes.py:
```diff
@@ -20,19 +20,24 @@ def getAttributeGetter(propName):
-        line = getAttributeSetterLine("owner", prop, "values[{}]".format(i))
-        code += "    " + line + "\n"
+        lines = getAttributeSetterLines("owner", prop, "values[{}]".format(i))
+        code += ''.join("    " + line + "\n" for line in lines)
```
Double quotes.

OmarEmaraDev commented on animation_nodes/utils/attributes.py:
```diff
@@ -48,16 +53,14 @@ def pathBelongsToArray(object, dataPath):
     try:
-        amount = eval("object{}.bl_rna.properties[{}].array_length".format(pathToProperty, repr(propertyName)))
-        return amount > 0
+        data = eval("object{}".format(dataPath))
```
Use f-strings.

OmarEmaraDev commented on animation_nodes/utils/attributes.py:
```diff
@@ -48,16 +53,14 @@ def pathBelongsToArray(object, dataPath):
-        return amount > 0
+        data = eval("object{}".format(dataPath))
+        if isinstance(data, str):
+            return False # Strings have len() but aren't arrays
```
Comments go above statements.

OmarEmaraDev reviewed (changes requested):
Looks generally fine. Just some style comments to follow AN's code style.

Th3Gavst3r commented:
Sorry for the delay on my end this time, lots of holidays lately.

Going from your examples I fixed up some other `f` strings in `attributes.py`, which made made realize several other functions were using the `object.` notation that caused #1813. I haven't had those functions cause me problems yet, but I went ahead and modified them.

I also added an `object.update_tag()` to the setters because Blender has a [known issue](https://developer.blender.org/T63793#881438) where custom properties when controlled by Python don't trigger updates to the dependency graph. This means drivers controlled by the value won't work unless you tag the object for updates.

OmarEmaraDev reviewed (approved):
Looks good. Thanks!
